vllm/profiler/layerwise_profile.py

```diff
@@ -229,6 +282,61 @@ def _total_cuda_time(self):
         return sum(
             [self._cumulative_cuda_time(root) for root in self._module_tree])
 
+    def _get_flop_count_for_layer(self, layer_name: str) -> int:
+        """Get FLOP count for a specific layer from flop_counts.
+        
+        Args:
+            layer_name: Name of the layer to get FLOP counts for.
+        """
+        if not self.flop_counts or not self.flop_counts.layer_counts:
+            return 0
+
+        # First try exact match
+        if layer_name in self.flop_counts.layer_counts:
+            return self.flop_counts.layer_counts[layer_name].total()
+
+        # Then try hierarchical prefix matching (for nested modules)
+        for layer, flop_count in self.flop_counts.layer_counts.items():
+            if layer_name.startswith(layer +
+                                     ".") or layer.startswith(layer_name +
+                                                              "."):
+                return flop_count.total()
+
+        return 0
+
+    def _get_flop_count_for_operation(self, op_name: str) -> int:
+        """Get FLOP count for a specific operation from flop_counts.
+        
+        Args:
+            op_name: Name of the operation to get FLOP counts for.
+        """
+        if not self.flop_counts or not self.flop_counts.operation_counts:
+            return 0
+
+        # First try exact match
+        if op_name in self.flop_counts.operation_counts:
+            return self.flop_counts.operation_counts[op_name]
+
+        # Then try suffix matching for operation overloads
+        for op, flops in self.flop_counts.operation_counts.items():
+            if op_name.endswith(op.split('.')[-1]) or op.endswith(
+                    op_name.split('.')[-1]):
+                return flops
+
+        return 0
+
+    def _calculate_gflops_per_sec(self, flops: int, time_us: float) -> float:
```

Comment:
why flops is int?

Reply:
The pytorch library returns ints for each operation. For example here: https://github.com/pytorch/pytorch/blob/main/torch/utils/flop_counter.py#L58